site/docs/hosting/comparison.md

```diff
@@ -18,17 +18,21 @@ In other words, you just run it on someone else's computer.
 
 We have two comparison tables: one for [serverless](#serverless) hosting and one for [VPS](#vps).
 
-### Serverless
+### Serverless and PaaS
 
 Serverless means that you do not control a single machine on which your bot is run.
 Instead, these hosting providers will rather allow you to upload your code, and then start and stop different machines as necessary to make sure that your bot always works.
 
-This has the downside that your bot does not have access to a persistent storage by default, such as a local file system.
-Instead, you will often have to have a database separately and connect to it if you need to store data permanently.
-We therefore recommend you to use a different kind of hosting for more complex bots, e.g. a [VPS](./vps.md).
-
 The main thing to know about them is that on serverless infrastructures you are required to use [webhooks](../guide/deployment-types.md).
```

Comment:
Most of providers below will have issues when you try running your bot with polling (`bot.start()` or grammY runner) on them.